### PR TITLE
Fix conf-nanomsg

### PR DESCRIPTION
**Pull request by vbmithr**

### CODE REVIEW

File: packages/conf-nanomsg/conf-nanomsg.0/opam [hunk 1]
```diff
@@ -1,12 +1,12 @@
 opam-version: "1"
-maintainer: "Rudi Grinberg"
+maintainer: "Vincent Bernardoff"
 authors: ["Martin Sustrik"]
 homepage: "http://nanomsg.org/"
 license: "MIT/X11"
-build: [["pkg-config" "libnanomsg"]]
+build: [["test" "-f" "/usr/include/nanomsg/nn.h"]]
```
Comment (avsm) [inline packages/conf-nanomsg/conf-nanomsg.0/opam]: This doesnt seem like it'll work outside a few Linux distros (what about Homebrew or FreeBSD in /usr/local)? /cc @rgrinberg as well.

**Comment by vbmithr** [comment]
On 23/12/2016 22:05, Anil Madhavapeddy wrote:
> This doesnt seem like it'll work outside a few Linux distros (what about
> Homebrew or FreeBSD in /usr/local)? /cc @rgrinberg
> <https://github.com/rgrinberg> as well.
> 

Agreed, the most generic solution is to try to compile something with #include
<nanomsg/nn.h> in the beginning. I pushed this fix because it worked for a "few
linux distros".

**Comment by rgrinberg** [comment]
Hmm, but what's the problem with using pkg-config?

Did nano change their installation not to use it or something?

**Comment by vbmithr** [comment]
On 23/12/2016 22:09, Rudi Grinberg wrote:
> Hmm, but what's the problem with using pkg-config?
> 
> Did nano change their installation not to use it or something?
> 

All distros don't use the same name for pkg-config.
On debian it's libnanomsg, on Arch nanomsg

**Comment by rgrinberg** [comment]
>All distros don't use the same name for pkg-config.
>On debian it's libnanomsg, on Arch nanomsg

Ahh, that's a bummer. Can we make something like this work perhaps:

$ pkg-config nanomsg || pkg-config libnanomsg

**Comment by dbuenzli** [comment]
> Ahh, that's a bummer.

And should be reported upstream. What's the point of `pkg-config` if everyone out there uses different package names for the same thing.

**Comment by vbmithr** [comment]
On 24/12/2016 09:06, Daniel Bünzli wrote:
> And should be reported upstream. What's the point of |pkg-config| if everyone
> out there uses different package names for the same thing.

I think this is because the Arch package is nanomsg 1.0 whereas the debian
jessie one is stuck to version 0.4. They might have changed the name oh the
pkg-config .pc file in the meantime.

The first thing to be clear about is which nanomsg version we are supporting.

**Comment by rgrinberg** [comment]
>I think this is because the Arch package is nanomsg 1.0 whereas the debian
>jessie one is stuck to version 0.4. They might have changed the name oh the
>pkg-config .pc file in the meantime.

If that's the case, I'd stick to supporting the 1.0 name (as long as it's
consistent).

**Comment by vbmithr** [comment]
On 24/12/2016 17:32, Rudi Grinberg wrote:
> If that's the case, I'd stick to supporting the 1.0 name (as long as it's
> consistent).

Ok, but constistent to what exactly? The 1.0 name is `nanomsg`

**Comment by rgrinberg** [comment]
So I think that settles it. Let's use the 1.0 name (`nanomsg`) and pull depext support for distros that haven't upgraded to a recent version yet.

**Pull request merged**